src/AutoMapper/QueryableExtensions/Impl/QueryMapperHelper.cs

```diff
@@ -34,5 +34,9 @@ public static TypeMap CheckIfMapExists(this IConfigurationProvider config, Type
 
         public static Exception MissingMapException(Type sourceType, Type destinationType) 
             => new InvalidOperationException($"Missing map from {sourceType} to {destinationType}. Create using Mapper.CreateMap<{sourceType.Name}, {destinationType.Name}>.");
+
+        public static Exception MissingPropertyMapException(string sourceFullName, Type sourceType, Type destinationType)
```

Comment:
The idea was to use the same helper above in GetPropertyMap.  And the message is misleading. I say throw AutoMapperConfigurationException. That seems to have the right message.